### DISCUSSION TITLE
RAND_priv_bytes return failure

### DISCUSSION THREAD

**Discussion by yalanmao** [openssl/openssl#25099] [Q&A] [answered]
linux os:Kylin Linux Advanced Server V10
cpu: Loongson-3A R4 (Loongson-3B4000)
openssl version: 1.1.1u
RAND_priv_bytes(buf,32) return failure,ERR_reason_error_string printf"error retrieving entropy", ERR_lib_error_string printf "random number generator" , ERR_func_error_string printf "RAND_DRBG_generate"
![2689D646-C2C0-42E3-8525-41F18D8D38B9](https://github.com/user-attachments/assets/ad77d6e6-2960-4b1e-b84a-7227071713af)

**Comment by t8m** [comment, marked as answer]
Unfortunately 1.1.1 version is out of support. Could you please try to reproduce with any 3.x version?

**Reply by yalanmao** [reply]
Sorry, we can only use version 1.1.1. The internal implementation of random number generation should be the same. I guess it is a problem with the operating system. Some methods are needed to troubleshoot the problem.

**Reply by t8m** [reply]
I'd suggest using strace to see what kernel syscalls fail/provide.

**Reply by yalanmao** [reply]
Thank you, I will let you know the results of the investigation

**Reply by paulidale** [reply]
The RNG system had fairly significant updates between 1.1.1 and 3.0.  So they aren't the same.
